docs/website/docs/building-from-source/riscv.md

```diff
@@ -0,0 +1,137 @@
+
+### Install Prebuilt RISC-V Tools (RISC-V 64-bit Linux toolchain)
+
+Execute the following script to download the prebuilt RISC-V toolchain and QEMU:
+
+```shell
+# In IREE source root
+$ ./build_tools/riscv/riscv_bootstrap.sh
+```
+**NOTE**:
+* You also need to set `RISCV_TOOLCHAIN_ROOT`
+(default at ${HOME}/riscv/toolchain/clang/linux/RISCV).
+
+## Configure and build
+
+### Host configuration
+
+Build and install on your host machine:
+
+``` shell
+cmake -B ../iree-build/ \
+  -DCMAKE_INSTALL_PREFIX=../iree-build/install \
+  -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+  .
+cmake --build ../iree-build/ --target install
+```
+
+### Target configuration
+
+Currently IREE verifies (and tests in CI) two RISC-V targets:
+* RISC-V 64-bit CPU with Linux OS
+* RISC-V 32-bit CPU with "bare-metal" config -- a limited subset of IREE runtime
+libraries that _can_ run on the bare-metal machine mode (with the machine BSP
+linker script), but can also run on larger machines without the system library
+support.
+
+For other RISC-V targets, please refer to
+[riscv.toolchain.cmake](https://github.com/google/iree/blob/main/build_tools/cmake/riscv.toolchain.cmake)
+as a reference of how to set up the cmake configuration.
+
+#### RISC-V 64-bit Linux target
+```shell
+$ cmake -G Ninja -B ../iree-build-riscv/ \
+  -DCMAKE_TOOLCHAIN_FILE="./build_tools/cmake/riscv.toolchain.cmake" \
+  -DIREE_HOST_BINARY_ROOT=$(realpath ../iree-build-host/install) \
+  -DRISCV_CPU=rv64 \
+  -DIREE_BUILD_COMPILER=OFF \
+  -DIREE_ENABLE_MLIR=OFF \
+  -DIREE_BUILD_SAMPLES=ON \
+  -DRISCV_TOOLCHAIN_ROOT=${RISCV_TOOLCHAIN_ROOT}
+  .
+```
+
+For the RISC-V 32-bit bare-metal config, append the following CMake options
+```shell
+-DRISCV_CPU=rv32-baremetal \
+-DIREE_BUILD_TESTS=OFF
+```
+
+## Running IREE bytecode modules on the RISC-V system
+
+**NOTE**:The following instructions are meant for the RISC-V 64-bit Linux
+target. For the bare-metal target, please refer to
+[simple_embedding](https://github.com/google/iree/blob/main/iree/samples/simple_embedding)
+to see how to build a ML workload for a bare-metal machine.
+
+Set the environment variable `RISCV_TOOLCHAIN_ROOT` if it is not set yet:
+
+```shell
+$ export RISCV_TOOLCHAIN_ROOT=<root directory of the RISC-V GNU toolchain>
+```
+
+Set the path to qemu-riscv64 Linux emulator binary in the `QEMU_BIN` environment
+variable. If it is installed with `riscv_bootstrap.sh`, the path is default at
+${HOME}/riscv/qemu/linux/RISCV/bin/qemu-riscv64.
+
+```shell
+$ export QEMU_BIN=<path to qemu-riscv64 binary>
+```
+
+Invoke the host compiler tools produce input files:
+
+``` shell
+$ ../iree-build/install/bin/iree-translate \
+  -iree-mlir-to-vm-bytecode-module \
+  -iree-hal-target-backends=vmvx \
+  iree/samples/models/simple_abs.mlir \
+  -o /tmp/simple_abs_vmvx.vmfb
+```
+
+Run the RISC-V emulation:
+
+```shell
+$ ${QEMU_BIN} \
+  -cpu rv64,x-v=true,x-k=true,vlen=256,elen=64,vext_spec=v1.0 \
```

Comment:
Are these options worth explaining a bit? They look like magic to me :P

Maybe there's a reference page that could be linked? For Android, [this section](https://google.github.io/iree/building-from-source/android/#target-configuration) links to two external guides that explain the arguments (like this: https://developer.android.com/ndk/guides/cmake#variables)

Reply:
Same situation w.r.t supporting the vector extension instructions. This is the corresponding part of enabling the vector extension from the emulator side.

Reply:
Drop the vector extension support per offline conversion.